File: producer_pact/src/main/java/com/example/intoxication/BeerServingController.java
package com.example.intoxication;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

/**
 * Scenario based controller
 */
@RestController
public class BeerServingController {

	private final ResponseProvider responseProvider;

	public BeerServingController(ResponseProvider responseProvider) {
		this.responseProvider = responseProvider;
	}

	@RequestMapping(value = "/beer",
			method=RequestMethod.POST,
			consumes="application/json",
			produces="application/json")
	public Response check(@RequestBody Customer customer) {
		//remove::start[]
		return this.responseProvider.thereYouGo(customer);
		//remove::end[return]
	}
	
}

interface ResponseProvider {
	Response thereYouGo(Customer personToCheck);
}

class Customer {
	//remove::start[]
	public String name;

	public Customer(String name) {
		this.name = name;
	}

	public Customer() {
	}
	//remove::end[]
}

class Response {
	public DrunkLevel previousStatus;
	public DrunkLevel currentStatus;

	public Response(DrunkLevel previousStatus, DrunkLevel currentStatus) {
		this.previousStatus = previousStatus;
		this.currentStatus = currentStatus;
	}
}

enum DrunkLevel {
	SOBER, TIPSY, DRUNK, WASTED
}

File: producer_kotlin/src/main/kotlin/com/example/intoxication/BeerServingController.kt
package com.example.intoxication

import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RestController

/**
 * Scenario based controller
 */
@RestController
class BeerServingController(private val responseProvider: ResponseProvider) {

    @RequestMapping(value = ["/beer"],
            method = arrayOf(RequestMethod.POST),
            consumes = arrayOf("application/json"),
            produces = arrayOf("application/json"))
    fun check(@RequestBody customer: Customer): Response {
        return this.responseProvider.thereYouGo(customer)
    }

}

interface ResponseProvider {
    fun thereYouGo(personToCheck: Customer): Response
}

class Customer {
    //remove::start[]
    var name: String = ""

    constructor(name: String) {
        this.name = name
    }

    constructor() {}
    //remove::end[]
}

class Response(var previousStatus: DrunkLevel, var currentStatus: DrunkLevel)

enum class DrunkLevel {
    SOBER, TIPSY, DRUNK, WASTED
}